packages/bsky/src/views/index.ts

```diff
@@ -453,6 +455,33 @@ export class Views {
     }
   }
 
+  status(did: string, state: HydrationState): StatusView | undefined {
+    const actor = state.actors?.get(did)
+    if (!actor?.status) return
+
+    const { status } = actor
+    const { record, sortedAt } = status
+
+    const maxDuration = 3 * HOUR
+    const expiresAtMs = record.durationMinutes
+      ? sortedAt.getTime() +
+        Math.min(record.durationMinutes * MINUTE, maxDuration)
+      : undefined
+    const expiresAt = expiresAtMs
+      ? new Date(expiresAtMs).toISOString()
+      : undefined
+
+    const isActive = expiresAtMs ? expiresAtMs > Date.now() : undefined
```

Comment:
I don't think we should necessarily do this as part of this work, but I am wondering if we should eventually add the current time into `HydrationState` so that all time-dependent view logic consistently hinges off the same time.